zebra-chain/src/transaction.rs

```diff
@@ -146,4 +154,33 @@ impl Transaction {
                 Some(TransparentInput::Coinbase { .. })
             )
     }
+
+    // TODO(jlusby): refine type
+    pub fn sighash(&self, network: Network, height: BlockHeight, hash_type: u32) -> Hash {
```

Comment:
Rather than having `hash_type: u32`, `hash_type` should be an enum of the possible sighash methods.

Reply:
the hash_type is a bunch of bigflags, we could convert it into an enum using https://docs.rs/bitflags/1.2.1/bitflags/